src/frequenz/quantities/marshmallow.py

```diff
@@ -0,0 +1,192 @@
+    Current,
+    Energy,
+    Frequency,
+    Percentage,
+    Power,
+    Temperature,
+    Voltage,
+]
+
+
+def _create_quantity_field_class(
+    quantity_subclass: Type[Quantity],
+) -> Type[fields.Field]:
+    """Dynamically create a QuantityField subclass for a given Quantity subclass."""
+    class_name = f"{quantity_subclass.__name__}Field"
+
+    field_class: Type[fields.Field] = type(
+        class_name,
+        (_QuantityField,),
+        {
+            "__init__": lambda self, **kwargs: super(field_class, self).__init__(
+                field_type=quantity_subclass, **kwargs
+            ),
+            "__module__": __name__,
+        },
+    )
+
+    return field_class
+
+
+QUANTITY_FIELD_CLASSES = {
+    quantity_subclass: _create_quantity_field_class(quantity_subclass)
+    for quantity_subclass in _QUANTITY_SUBCLASSES
+}
+"""Mapping of Quantity subclasses to their corresponding QuantityField subclasses."""
+
+
+class QuantitySchema(Schema):
+    """A schema for quantities.
+
+    Example usage:
+
+    ```python
+    from dataclasses import dataclass, field
+    from marshmallow_dataclass import class_schema
+    from marshmallow.validate import Range
+    from frequenz.quantities import Percentage, QuantitySchema
+    from typing import cast
+
+    @dataclass
+    class Config:
+        percentage_always_as_string: Percentage = field(
+            default_factory=lambda: Percentage.from_percent(25.0),
+            metadata={
+                "metadata": {
+                    "description": "A percentage field",
+                },
+                "validate": Range(Percentage.zero(), Percentage.from_percent(100.0)),
+                "serialize_as_string": True,
+            },
+        )
+
+        percentage_always_as_float: Percentage = field(
+            default_factory=lambda: Percentage.from_percent(25.0),
+            metadata={
+                "metadata": {
+                    "description": "A percentage field",
+                },
+                "validate": Range(Percentage.zero(), Percentage.from_percent(100.0)),
+                "serialize_as_string": False,
+            },
+        )
+
+        percentage_serialized_as_schema_default: Percentage = field(
+            default_factory=lambda: Percentage.from_percent(25.0),
+            metadata={
+                "metadata": {
+                    "description": "A percentage field",
+                },
+                "validate": Range(Percentage.zero(), Percentage.from_percent(100.0)),
+            },
+        )
+
+        @classmethod
+        def load(cls, config: dict[str, Any]) -> "Config":
+            schema = class_schema(cls, base_schema=QuantitySchema)(
+                serialize_as_string_default=True # type: ignore[call-arg]
+            )
+            return cast(Config, schema.load(config))
+    ```
+    """
+
+    class Meta:
+        """Meta information."""
+
+        unknown = EXCLUDE
+        """Ignore unknown fields."""
```

Comment:
Why this? I wouldn't force this option on all subclasses.

Reply:
Clearly I would :P
But it's just the default, you can always pass it as parameter.
I can also remove it

Reply:
Yeah, this option is totally unrelated to the quantities, I would not change the default here.